utils/controller_debug.py

```diff
@@ -361,222 +380,288 @@ def peek32( addr, ct=None, le=True, signed=False ):
-def Split32( x, le=True, asStr=False ): return SplitInt( x, 4, le, asStr );
-def SplitFlt( x ): return Split32( [F2I(i) for i in x] )
+    cmd = [TERM, TERM]
+    ser.write("".join([chr(x) for x in cmd]))
+    time.sleep(0.1)
+    ser.reset_input_buffer()
+
+
+def I2F(ival):
+    s = struct.pack("I", ival & 0xFFFFFFFF)
+    return struct.unpack("f", s)[0]
+
+
+def F2I(fval):
+    s = struct.pack("f", fval)
+    return struct.unpack("I", s)[0]
+
+
+def SplitInt(val, len=4, le=True, asStr=False):
+    ret = []
+
+    if isinstance(val, list):
+        for v in val:
+            ret += SplitInt(v, len, le, False)
+    else:
+        for i in range(len):
+            ret.append(int(val & 0x00FF))
+            val >>= 8
+
+        if not le:
+            ret = ret[::-1]
+    if asStr:
+        ret = "".join([chr(i) for i in ret])
+    return ret
+
+
+def Split16(x, le=True, asStr=False):
+    return SplitInt(x, 2, le, asStr)
+
+
+def Split32(x, le=True, asStr=False):
+    return SplitInt(x, 4, le, asStr)
+
+
+def SplitFlt(x):
+    return Split32([F2I(i) for i in x])
+
 
 # Utility function to convert bytes to ints
-def MakeInt( bytes, signed=True, le=True ):
-   if( not le ): bytes = bytes[::-1];
-   val = 0;
-   shift = 0;
-   for b in bytes:
-      val |= (b<<shift);
-      shift += 8;
-
-   if( signed and bytes[-1] & 0x80 ):
-      return val - (1<<shift)
-   return val
-
-def Build32( dat, le=True, signed=False ):
-   ret = [];
-   for i in range( len(dat)/4 ):
-      ret.append( MakeInt( dat[ (4*i):(4*i+4) ], signed=signed, le=le ) );
-   return ret;
-
-def Build16( dat, le=True, signed=False ):
-   ret = [];
-   for i in range( len(dat)/2 ):
-      ret.append( MakeInt( dat[ (2*i):(2*i+2) ], signed=signed, le=le ) );
-   return ret;
-
-def BuildFlt( dat ):
-   tmp = Build32( dat, le=True, signed=False )
-   return [I2F(x) for x in tmp]
-
-def GrabElems( dat, ct ):
-   if( len(dat) < ct ):
-      dat.extend( (ct-len(dat))*[0] );
-   ret = dat[:ct];
-   del dat[0:ct];
-   return ret;
-
-def GrabI8( dat, le=True ):  return MakeInt( GrabElems(dat,1), signed=True,  le=le );
-def GrabU8( dat, le=True ):  return MakeInt( GrabElems(dat,1), signed=False, le=le );
-def GrabI16( dat, le=True ): return MakeInt( GrabElems(dat,2), signed=True,  le=le );
-def GrabU16( dat, le=True ): return MakeInt( GrabElems(dat,2), signed=False, le=le );
-def GrabI24( dat, le=True ): return MakeInt( GrabElems(dat,3), signed=True,  le=le );
-def GrabU24( dat, le=True ): return MakeInt( GrabElems(dat,3), signed=False, le=le );
-def GrabI32( dat, le=True ): return MakeInt( GrabElems(dat,4), signed=True,  le=le );
-def GrabU32( dat, le=True ): return MakeInt( GrabElems(dat,4), signed=False, le=le );
-def GrabI64( dat, le=True ): return MakeInt( GrabElems(dat,8), signed=True,  le=le );
-def GrabU64( dat, le=True ): return MakeInt( GrabElems(dat,8), signed=False, le=le );
-def GrabFlt( dat, le=True ): return I2F( GrabU32(dat, le=le) );
+def MakeInt(bytes, signed=True, le=True):
```

Comment:
I think this is what the Python https://docs.python.org/3.7/library/struct.html module is for.

(Alternatively we could use a protobuf for this.)

Reply:
I use the struct module elsewhere, but this has the side effect of pulling the first N bytes off the passed list.  It's convenient for what it's used for

Reply:
Don't quite understand which side effect you mean? Note that another alternative to struct is https://docs.python.org/3.2/library/stdtypes.html#int.from_bytes which has exactly the API of the current function.